myTaskBoard.js

```diff
@@ -0,0 +1,136 @@
+    const paragraph = document.createElement("p");
+    paragraph.classList.add("taskText", "float-start", "col-lg-10");
+    paragraph.innerText = task.taskIntro;
+    function createDeleteButton() {
+        const deleteButton = document.createElement("button");
+        const iconX = document.createElement("i");
+        iconX.classList.add("bi", "bi-x-lg", "iconSize");
+        deleteButton.classList.add("deleteButton", "float-end", "col-lg-2")
+        deleteButton.append(iconX);
+
+
+        deleteButton.addEventListener("click", function () {
+            if (typeof task.taskId !== "string") { return }
+            const id = task.taskId;
+            for (let index = 0; index < state.MyTasks.length; index++) {
+                const currentTask = state.MyTasks[index];
+                if (currentTask.taskId === id) {
+                    state.MyTasks.splice(index, 1)
+                    console.log(`i found the movie`);
+                    const tasksString = localStorage.getItem("MyTasks");
+                    let tasks = JSON.parse(tasksString);
+                    tasks = state.MyTasks;
+                    localStorage.setItem("MyTasks", JSON.stringify(tasks));
+                    draw(state.MyTasks)
+                }
+
+            }
+
+
+        })
+        return deleteButton
+    }
+    const deleteButton = createDeleteButton();
+
+    upperDiv.append(paragraph, deleteButton);
+    upperDiv.classList.add("noteUpperDiv")
+    outerDiv.addEventListener("mouseenter", function () {
+        deleteButton.style.visibility = "visible"
+    });
+    outerDiv.addEventListener("mouseleave", function () {
+        deleteButton.style.visibility = "hidden";
+    })
+    if (task.taskIntro.length > 30) { upperDiv.classList.add("noteUpperDivScroll") }
+
+    const lowerDiv = document.createElement("div");
+    const dateSpan = document.createElement("span");
+    const timeSpan = document.createElement("span");
+
+    function changeDateDisplay() {
+        const d = new Date(task.taskDate);
+        const year = d.getFullYear();
+        const month = d.getMonth() + 1;
+        const day = d.getDay();
+        const dateText = `${day}/${month}/${year}`
+
+        return dateText
+
+    }
+
+    dateSpan.innerHTML = changeDateDisplay();
+    timeSpan.innerHTML = task.taskTime;
+    lowerDiv.append(dateSpan, timeSpan);
+    lowerDiv.classList.add("noteLowerDiv")
+    outerDiv.append(upperDiv, lowerDiv)
+    outerDiv.classList.add("note", "col-lg-3", "fadein");
+    return outerDiv;
+}
+function createNote() {
+    const task = {
+        taskIntro: DOM.textAreaRef.value,
+        taskDate: DOM.dateRef.value,
+        taskTime: DOM.timeRef.value,
+        taskId: `${Math.ceil(Math.random() * 999)}`,
+    }
+    if (task.taskIntro.length <= 0) {
+        return alert(`Insert Task Details`)
+    }
+    if (task.taskDate.length > 10) {
+        return alert(`Insert Logical Date Please !`)
+    }
+    if (task.taskDate === "NaN/NaN/NaN") {
```

Comment:
you can unite between the 
and checking if a date is valid could be better with Date(testDateValidation)
and receive back Invalid Date
for example: 
<img width="348" alt="Screen Shot 2022-01-23 at 0 12 09" src="https://user-images.githubusercontent.com/32986703/150657083-f66ee34f-e515-432b-bf1f-4bde769bbd84.png">